R/sitrep.R

```diff
@@ -2,17 +2,31 @@
 #' @importFrom memoise memoise
 NULL
 
-check_for_rstudio_updates <- function(os = tolower(Sys.info()[["sysname"]]), version = rstudioapi::getVersion(), in_rstudio = rstudioapi::isAvailable()) {
-
+check_for_rstudio_updates <- function(os = tolower(Sys.info()[["sysname"]]),
+                                      version = rstudioapi::versionInfo()$long_version,
+                                      in_rstudio = rstudioapi::isAvailable()) {
   if (!in_rstudio) {
     return()
   }
 
-  url <- sprintf("https://www.rstudio.org/links/check_for_update?version=%s&os=%s&format=%s", version, os, "kvp")
+  url <- sprintf(
+    "https://www.rstudio.org/links/check_for_update?version=%s&os=%s&format=%s",
+    utils::URLencode(version, reserved = TRUE), os, "kvp"
+  )
 
   tmp <- file_temp()
-  on.exit(file_delete(tmp))
-  utils::download.file(url, tmp, quiet = TRUE)
+  withr::defer(file_exists(tmp) && nzchar(file_delete(tmp)))
+  suppressWarnings(
+    download_ok <- tryCatch({
+      utils::download.file(url, tmp, quiet = TRUE)
+      TRUE
+    }, error = function(e) FALSE)
+  )
+  if (!download_ok) {
+    # I'll take silent failure here over dev_sitrep() falling over completely
+    # if this download fails
```

Comment:
This captures why I've made this function a bit more robust. In general, I think "sitrep" functions should basically never throw an error, because you're often using them to bring clarity to weird, sub-optimal situations.